src/Components/Components/src/ChangeDetection.cs

```diff
@@ -41,6 +44,24 @@ public static bool MayHaveChanged<T1, T2>(T1 oldValue, T2 newValue)
     // For performance reasons, the following immutable types are not supported: IntPtr, UIntPtr, Type.
     private static bool IsKnownImmutableType(Type type)
         => Type.GetTypeCode(type) != TypeCode.Object
-        || type == typeof(Guid)
+        || _immutableNonObjectTypesCache.GetOrAdd(type, IsImmutableNonObjectCore);
+
+    private static bool IsImmutableNonObjectCore(Type type)
+        => type == typeof(Guid)
+        || type == typeof(DateOnly)
+        || type == typeof(TimeOnly)
         || typeof(IEventCallback).IsAssignableFrom(type);
+
+    public static event Action? OnHotReload;
+
+    public static void ClearImmutableTypeCache()
+    {
+        _immutableNonObjectTypesCache.Clear();
+    }
+
+    public static void HotReloadHandler()
+    {
+        ClearImmutableTypeCache();
+        OnHotReload?.Invoke();
+    }
```

Comment:
It doesn't look like this hooks into the hot reload `OnDeltaApplied` event, so the cachce will never get cleared.

This can all probably be reduced down to something like what we do for RouteView: https://github.com/dotnet/aspnetcore/blob/870eec223cdb09e5ca8cfdc61bfd5fbc2ca9773b/src/Components/Components/src/RouteView.cs#L23-L29